src/main/java/pl/task/parkingmeter/controller/VehicleController.java

```diff
@@ -0,0 +1,205 @@
+
+        if (matcher.matches()) {
+
+            LocalDateTime[] dates = returnStartAndEndDate(date);
+            List<String> allCurrency = vehicleService.getAllCurrency();
+
+            allCurrency.stream().forEach(currency -> {
+                Profit profit = new Profit();
+                profit.setDate(dates[1].toLocalDate());
+                profit.setCurrency(currency);
+                profit.setValue(countProfitForCurrency(dates, currency));
+                listOfProfit.add(profit);});
+
+        } else {
+            throw new InvalidDateException(date);
+        }
+        return listOfProfit;
+    }
+
+    private BigDecimal countProfitForCurrency(LocalDateTime[] dates, String currency) {
+
+        List<Vehicle> vehicles = vehicleService.findVehiclesByPayDateBetweenAndCurrency(dates[0], dates[1], currency);
+        BigDecimal profit = new BigDecimal(0);
+
+        if (!vehicles.isEmpty() || vehicles != null) {
+            for (Vehicle vehicle : vehicles) {
+                profit = profit.add(vehicle.getBill());
+            }
+        }
+        return profit;
+    }
+
+    private LocalDateTime[] returnStartAndEndDate(String date) {
+
+        LocalDateTime[] dates = new LocalDateTime[2];
+
+        String[] parsedDate = date.split("-");
+
+        dates[0] = LocalDateTime
+                .of(Integer.valueOf(parsedDate[0]), Integer.valueOf(parsedDate[1]), Integer.valueOf(parsedDate[2]), 0, 0, 0);
+
+        dates[1] = LocalDateTime
+                .of(Integer.valueOf(parsedDate[0]), Integer.valueOf(parsedDate[1]), Integer.valueOf(parsedDate[2]), 23, 59, 59);
+
+        return dates;
+    }
+
+    private String checkRegNumber(String regNumber) {
+
+        regNumber = regNumber.replaceAll(" ", "").toUpperCase();
+
+        Matcher matcher = PATTERN_REGNUMBER.matcher(regNumber);
+
+        if (matcher.matches()) {
+            return regNumber;
+        } else {
+            throw new InvalidRegNumberException(regNumber);
+        }
+    }
+
+    private long checkHours(LocalDateTime createdDate) {
+
+        LocalDateTime startTime = createdDate;
+        Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+        LocalDateTime actualTime = timestamp.toLocalDateTime();
+
+        return ChronoUnit.HOURS.between(startTime, actualTime) + 1;
+    }
+
+    private BigDecimal getValueToPay(String currency, Vehicle vehicle) {
+
+        long hours = checkHours(vehicle.getRunDate());
+        BigDecimal valueToPay;
+
+        if (hours > 24) {
+            valueToPay = BigDecimal.valueOf(5000);
+        } else {
+            String type;
+
+            if (vehicle.isOwnerDisabled()) {
+                type = "disabled";
+            } else {
+                type = "regular";
+            }
+
+            Rates rates = ratesService.findRatesByTypeAndCurrency(type, currency);
+            valueToPay = valueRateService.findValueRateByHoursAndRatesId(hours, rates).getValue();
+
```

Comment:
Ciekawie rozwiązanie zwracania kosztu parkingu.

1. Przyjąłeś, że za parkowanie dłużej niż 24h płaci się stałą kwotę. Nie było tego nigdzie w poleceniu zadania, ani nigdzie nie napisałeś takich założeń. Myślisz, że klient byłby zadowolony z takiego rozwiązania?
2. Zamiast wyliczania ad-hoc wyliczyłeś coś na początku (w jaki sposób? na kalkulatorze?) i wpisałeś konfigurację do bazy danych. Jakakolwiek zmiana algorytmu wyliczania ceny spowodowałaby, że znowu musiałbyś przeliczyć ceny dla wszystkich godzin i zupdate'ować je w bazie. Jeżeli byłby do tego automatyczny mechanizm, a to byłaby jedynie optymalizacja odczytu - to super. Wtedy jednak możnaby dodać cache'owanie tych danych.
3. Dodanie kolejnej waluty powoduje, że kolejne rekordy musisz umieścić w bazie danych... znowu na kalkulatorze.